### PR TITLE
[Models] add fleet model fallback 2

### PR DESCRIPTION
**Pull request by xiaoguoguo626807**
## Motivation
新增 PaddleFleet 作为模型推理后端（`--model-impl paddlefleet`），通过将 PaddleFleet TransformerLayer 中的 `core_attention` 替换为 FastDeploy Attention 内核，实现在 PaddleFleet 模型结构上复用 FastDeploy 的 KV Cache 和高性能 Attention 计算。 https://github.com/PaddlePaddle/FastDeploy/pull/7732 的清晰版

## Modifications
- `config.py`: 新增 `paddlefleet` 到 `ModelImpl` 类型定义
- `engine/args_utils.py`: 支持 `--model-impl paddlefleet` CLI 参数，并补充校验逻辑
- `model_executor/models/paddleformers/base_fleet.py`: 新增 `PaddleFleetModelBase` 基类、`FastDeployAttention` 层及 `patch_paddlefleet_core_attention` 替换函数
- `model_executor/models/paddleformers/__init__.py`: 注册 `PaddleFleetForCausalLM` 模型类
- test_fallback_fleet_model.py` 需要独立的 PaddleFormers 和 PaddleFleet 依赖,使用 pytest conftest.py 钩子机制，在测试运行时动态安装依赖，避免污染全局环境

## Usage or Command
```bash
python -m fastdeploy.entrypoints.openai.api_server \
    --model /path/to/model \
    --model-impl paddlefleet
```

## Accuracy Tests
N/A（本 PR 新增 PaddleFleet 推理后端，尚未提供与参考实现的 logits 对齐数据）

## Checklist

- [x] Add at least a tag in the PR title.
  - Tag list: [`[FDConfig]`,`[APIServer]`,`[Engine]`, `[Scheduler]`, `[PD Disaggregation]`, `[Executor]`, `[Graph Optimization]`, `[Speculative Decoding]`, `[RL]`, `[Models]`, `[Quantization]`, `[Loader]`, `[OP]`, `[KVCache]`, `[DataProcessor]`, `[BugFix]`, `[Docs]`, `[CI]`, `[Optimization]`, `[Feature]`, `[Benchmark]`, `[Others]`, `[XPU]`, `[HPU]`, `[GCU]`, `[DCU]`, `[Iluvatar]`, `[Metax]`]
  - You can add new tags based on the PR content, but the semantics must be clear.
- [x] Format your code, run `pre-commit` before commit.
- [x] Add unit tests. Please write the reason in this PR if no unit tests.
- [x] Provide accuracy results.
- [x] If the current PR is submitting to the `release` branch, make sure the PR has been submitted to the `develop` branch, then cherry-pick it to the `release` branch with the `[Cherry-Pick]` PR tag.

### CODE REVIEW

File: fastdeploy/model_executor/models/paddleformers/base_fleet.py [hunk 1]
```diff
@@ -0,0 +1,741 @@
...
+            #     self.paddleformers_config.tensor_model_parallel_size = 1
+            #     logger.warning("When using expert parallelism and tensor parallelism, sequence parallelism must be used in fleet set tp=1 .")
+            self.paddleformers_config.parallel_output = self.paddleformers_config.tensor_model_parallel_size == 1
+            self.paddleformers_config.max_seq_len = self.model_config.max_model_len
+            self.paddleformers_config.params_dtype = self.model_config.dtype or "bfloat16"
+            # self.paddleformers_config.moe_grouped_gemm = True
+            self.paddleformers_config.moe_token_dispatcher_type = "deepep"
+            # self.paddleformers_config.use_cpu_initialization = True
+            self.paddleformers_config.gated_attention = getattr(self.paddleformers_config, "use_gated_attn", False)
+            if self.paddleformers_config.multi_latent_attention:
```
[the remark this answers is not in the document]
Comment (xiaoguoguo626807) [reply fastdeploy/model_executor/models/paddleformers/base_fleet.py]: 默认有值

**Review by gongshaotian — APPROVED** [review]
LGTM for GraphOptBackend

**Review by xyxinyang — APPROVED** [review]
